sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/Config/IServiceManagerStore.cs

```diff
@@ -1,9 +1,11 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+
 namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
 {
-    internal interface IServiceManagerStore
+    internal interface IServiceManagerStore : IAsyncDisposable, IDisposable
```

Comment:
@pakrym do you know if ctrl + c is expected to trigger the DisposeAsync path? From testing, @Y-Sindo found that it only triggered Dispose, which is why she implemented both.

Reply:
What ASP.NET Core version?

Reply:
Is this type being registered with DI? DI does support AsyncDIsposables but it will only call DiposeAsync if DisposeAsync was called on the container. So picking between sync or async depends on the entire call chain during the cancellation process.

Reply:
@pakrym Yes, the type is registered with DI. See https://github.com/Azure/azure-sdk-for-net/pull/25341/files#diff-cf94d337234783f85317d7599ab410079e250c8b313bd7ef9fca0df41e98b352R30

Reply:
In the case we were looking at, ctrl + c was used to trigger shut down of the app.

Reply:
@JoshLove-msft the easiest way to know is to look at the stack trace. I don't know how CTRL+C is wired up in this particular environment.

Reply:
@pakrym We can see from the call stack that `WebHost.StopAsync` is called.
<details>
<summary>snapshots of call stack</summary>

![image](https://user-images.githubusercontent.com/15143836/142969985-3ffcf30b-bba9-4ca4-936e-f6f85e6d6122.png)

![image](https://user-images.githubusercontent.com/15143836/142970394-f6ddf64d-c9e9-4d51-bf75-fee116c241ca.png)

</details>